src/cosmicqc/image.py

```diff
@@ -0,0 +1,66 @@
+"""
+Helper functions for working with images in the context of coSMicQC.
+"""
+
+import cv2
+import numpy as np
+from PIL import Image, ImageEnhance
+
+
+def is_image_too_dark(image: Image, threshold: float = 10.0) -> bool:
+    """
+    Check if the image is too dark based on the mean brightness.
+    By "too dark" we mean not as visible to the human eye.
+
+    Args:
+        image (Image):
+            The input PIL Image.
+        threshold (float):
+            The brightness threshold below which the image is considered too dark.
+
+    Returns:
+        bool:
+            True if the image is too dark, False otherwise.
+    """
+    # Convert the image to a numpy array and then to grayscale
+    img_array = np.array(image)
+    gray_image = cv2.cvtColor(img_array, cv2.COLOR_RGBA2GRAY)
+
+    # Calculate the mean brightness
+    mean_brightness = np.mean(gray_image)
+
+    return mean_brightness < threshold
+
+
+def adjust_image_brightness(image: Image) -> Image:
+    """
+    Adjust the brightness of an image using histogram equalization.
+
+    Args:
+        image (Image):
+            The input PIL Image.
+
+    Returns:
+        Image:
+            The brightness-adjusted PIL Image.
+    """
+    # Convert the image to numpy array and then to grayscale
+    img_array = np.array(image)
+    gray_image = cv2.cvtColor(img_array, cv2.COLOR_RGBA2GRAY)
+
+    # Apply histogram equalization to improve the contrast
+    equalized_image = cv2.equalizeHist(gray_image)
+
+    # Convert back to RGBA
+    img_array[:, :, 0] = equalized_image  # Update only the R channel
+    img_array[:, :, 1] = equalized_image  # Update only the G channel
+    img_array[:, :, 2] = equalized_image  # Update only the B channel
```

Comment:
Thanks and great question about this - this happens prior to the the segmentation mask outline overlay on the image. The original image is treated as `RGBA` to prepare for a colored outline overlay (which otherwise would be treated as grayscale, without color). We do a quick conversion to grayscale here for the sake of brightness treatment, but it's not a default (some images might not need this treatment). As a result, the `RBGA` read of the original image feels accurate because we may be able to perform the colored outline overlay without further modifications.